coalib/settings/Section.py

```diff
@@ -161,6 +164,12 @@ def __getitem__(self, item, ignore_defaults=False):
 
         res = self.contents.get(key, None)
         if res is not None:
+            if res.to_append and self.defaults:
+                if res.key in self.defaults:
+                    return Setting(
+                        key=key,
+                        value=self.defaults[key].value + ',' + res.value,
+                        origin=res.origin)
```

Comment:
so there's an important case missing that should also be tested: how does writing back to file behave? I think it would write back the full concatenated setting.

Maybe we can build a mechanism that for every setting it checks if it `.startswith` it's default setting and the comma - and if so, strip that away and use `+=` automatically, then it'd even optimize it if I write coafiles that could profit from the `+=` operator but the user didn't use it.